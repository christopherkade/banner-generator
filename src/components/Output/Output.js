import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const Wrapper = styled.div`
  width: 1000px;
  height: 420px;
  background-color: ${(props) => props.bgColor};
  display: flex;
  justify-content: center;
  align-items: center;
  color: ${(props) => props.titleColor};
  border: 1px solid #dbdbdb;
  border: ${(props) => `${props.borderSize}px solid ${props.borderColor}`};
  box-sizing: border-box;

  @media (max-width: 1200px) {
    width: auto;
    height: auto;
  }
`

const Title = styled.h1`
  font-weight: 300;
  font-size: ${ (props) => props.size}rem;
`

const Output = ({ values }) => {
  const { bgColor, title, titleSize, titleColor, borderColor, borderSize } = values

  return (
    <Wrapper bgColor={bgColor} titleColor={titleColor} borderColor={borderColor} borderSize={borderSize} id="capture">
      <Title size={titleSize}>{title}</Title>
    </Wrapper>
  )
}

Output.propTypes = {
  values: PropTypes.shape({
    bgColor: PropTypes.string,
    title: PropTypes.string,
    titleSize: PropTypes.string,
    titleColor: PropTypes.string,
    borderColor: PropTypes.string,
    hasBorder: PropTypes.bool
  }).isRequired
}

export default Output
