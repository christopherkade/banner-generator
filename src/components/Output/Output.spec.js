import React from 'react'
import { render } from 'react-testing-library'

import Output from "./Output"

describe("<Output />", () => {
  const values = {
    bgColor: "black",
    title: "mockedTitle",
    titleSize: "5",
    titleColor: "white",
    borderSize: 0
  }

  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<Output values={values} />);
      expect(container).toMatchSnapshot();
    });
  })

  describe("title prop", () => {
    it("should display the title passed as prop", () => {
      const { container } = render(<Output values={values} />)
      expect(container.textContent).toContain("mockedTitle")
    })
  })
})
