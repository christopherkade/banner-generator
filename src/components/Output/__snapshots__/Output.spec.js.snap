// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Output /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa kbQBMw"
    id="capture"
  >
    <h1
      class="sc-bwzfXH cTtmfd"
      size="5"
    >
      mockedTitle
    </h1>
  </div>
</div>
`;
